src/kit.ts

```diff
@@ -580,6 +595,70 @@ async function collectDevBatVars(devbat: string, args: string[], major_version:
   return vars;
 }
 
+/**
+ * Gets the environment variables set by a shell script.
+ * @param kit The kit to get the environment variables for
+ */
+export async function getShellScriptEnvironment(kit: Kit): Promise<Map<string, string>|undefined> {
+  console.assert(kit.environmentVariablesShellScript);
+  const fname = Math.random().toString() + (process.platform == 'win32' ? '.bat' : '.sh');
+  const script_fname = `vs-cmt-${fname}`;
+  const envfname = script_fname + '.env';
+  const script_path = path.join(paths.tmpDir, script_fname);
+  const env_path = path.join(paths.tmpDir, envfname); // path of temp file in which the script writes the env vars to
+  const args = kit.environmentVariablesShellScriptArgs ? kit.environmentVariablesShellScriptArgs.map(arg => `"${arg}"`).join(" ") : "";
+
+  const script = [];
+  if (process.platform == 'win32') { // windows
+    if (kit.environmentVariablesShellScriptDirectory) {
+      script.push(`cd /d "${kit.environmentVariablesShellScriptDirectory}"`); // switch to directory if set
+    }
+    script.push(`call "${kit.environmentVariablesShellScript}" ${args}`); // call the user batch script
+    script.push(`set >> ${env_path}`); // write env vars to temp file
+    }
+  else { // linux based
+    if (kit.environmentVariablesShellScriptDirectory) {
+      script.push(`cd "${kit.environmentVariablesShellScriptDirectory}"`); // switch to directory if set
+    }
+    script.push(`. "${kit.environmentVariablesShellScript}" ${args}`); // run the user shell script
+    script.push(`printenv >> ${env_path}`); // write env vars to temp file
+  }
+  try {
+    await fs.unlink(env_path); // delete the temp file if it exists
+  } catch (error) {}
+  await fs.writeFile(script_path, script.join('\r\n')); // write batch file
+
+  const run_command = (process.platform == 'win32' ? 'call ' : '. ') + script_path;
+  const res = await proc.execute(run_command, [], null, {shell: true, silent: true}).result; // run script
+  await fs.unlink(script_path); // delete script file
+  const output = (res.stdout) ? res.stdout + (res.stderr || '') : res.stderr;
+
+  let env = '';
+  try {
+    /* When the script failed, envpath would not exist */
+    env = await fs.readFile(env_path, {encoding: 'utf8'});
+    await fs.unlink(env_path);
+  } catch (error) { log.error(error); }
+  if (!env || env === '') {
+    console.log(`Error running ${kit.environmentVariablesShellScript} with:`, output);
+    return;
+  }
+
+  // split and trim env vars
+  const vars
+      = env.split('\n').map(l => l.trim()).filter(l => l.length !== 0).reduce<Map<string, string>>((acc, line) => {
+          const mat = /(\w+)=?(.*)/.exec(line);
```

Comment:
We prefer it if you don't abbreviate variable names.  I think you mean `match` here.

Reply:
Okay, I fixed this